### ISSUE TITLE
Faster json parsing

### ISSUE THREAD

**Issue by ZJaume** [hplt-project/warc2text-runner#1]
[orjson](url) is[ several times faster](https://github.com/ijl/orjson#performance) than `ujson` or `json` from the standard library, and it is a drop-in replacement.

https://github.com/hplt-project/warc2text-runner/blob/f95ff23ee2f52a508abca8960f8fce86cf5ee1d6/two/trafilatura/traf.py#L4-L4

https://github.com/hplt-project/warc2text-runner/blob/f95ff23ee2f52a508abca8960f8fce86cf5ee1d6/two/fastertext_lid/proto_langid.py#L7

**Comment by nvanva** [comment]
Moved to orjson.

**Comment by nvanva** [comment]
The problem is orjson.dumps() is not drop-in replacement. Unlike json and ujson, its dumps() method returns a byte string:
![image](https://github.com/hplt-project/warc2text-runner/assets/9987540/2abf4c41-b33c-4973-92b2-c261c82a8b37)

I've made a comparison of speed, orjson is really faster, esp. its dumps() method.
![image](https://github.com/hplt-project/warc2text-runner/assets/9987540/3179ec34-2ae3-495f-8288-4748c7f91104)

![image](https://github.com/hplt-project/warc2text-runner/assets/9987540/45771679-ad4e-42a5-b57c-127a3290fc71)

orjson.loads() is 20% faster than ujson.loads() , but orjson.dumps() is >4x faster than ujson.dumps().

**Comment by nvanva** [comment]
The final performance of traf.py (and thus stage 2) will differ by less than 5%. The first one run with ```import orjson as json```, the second is with ```import ujson as json``` in traf.py:
![image](https://github.com/hplt-project/warc2text-runner/assets/9987540/346d050f-5cbf-4192-aee5-9b3d7e77db41)

**Comment by nvanva** [comment]
For now I've reverted to using ujson instead of orjson, orjson is not an in-place replacement because its dumps() returns a byte string instead of a string, so requires modification of printing it to stdout. Then we should change that and test again on 1% of all data, which seems to be not worth the potential speedup.

**Comment by ZJaume** [comment]
Yes, my bad, I didn't remember that byte thing. Maybe not a drop-in replacement because of that and serializing numpy float from fasttext requires enabling an option:
```
json_bytes = orjson.dumps(
        {'lang': self._postprocess_prediction(prediction), 'prob': round(prediction[1][0], 4)},
        option=orjson.OPT_SERIALIZE_NUMPY)
```

but serializing to bytes, in my opinion, it is just fine, just skip the additional string decoding step that others are doing:
```
sys.stdout.buffer.write(json_bytes)
sys.stdout.buffer.write(b'\n')
```

But also, it has another advantage, that is non-ascii characters not being escaped, which allow reading all non-english, non-latin languages directly without having to use `jq`.

**Comment by nvanva** [comment]
Loading / dumping jsonl takes a small fraction of time, further optimization of this won't result in a significant speedup.
